sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +385,352 @@ def transform(self, X):
+        - "roman" - Left to right.
+        - "arabic" - Right to left.
+        - "random" - A random order for each round.
+
+    n_imputations : int, optional (defaults = 100)
+        Number of MICE rounds to perform the results of which will be
+        used in the final average.
+
+    n_burn_in : int, optional (default = 10)
+        Number of initial MICE rounds to perform the results of which
+        will not be returned.
+
+    n_nearest_features : int, optional (default = np.infty)
+        Number of other features to use to estimate the missing values of
+        the current features. Can provide significant speed-up
+        when the number of features is huge.
+
+    initial_fill_method : str, optional (default = "mean" )
+        Valid values: {"mean", "median", or "most_frequent"}.
+        Uses :class:sklearn.preprocessing.Imputer.
+
+    min_value : float (default = -np.infty)
+        Minimum possible imputed value.
+
+    max_value : float (default = np.infty)
+        Maximum possible imputed value.
+
+    verbose : boolean (default = False)
+        Whether to provide updates on progress.
+
+    random_state : int seed, RandomState instance (default = None)
+        The seed of the pseudo random number generator to use when
+        shuffling the data.
+
+    Notes
+    -----
+    Features which only contain missing values at `fit` are discarded
+        upon `transform`.
+
+    References
+    ----------
+    .. [1] `Stef van Buuren, Karin Groothuis-Oudshoorn (2011). "mice:
+        Multivariate Imputation by Chained Equations in R". Journal of
+        Statistical Software 45: 1-67.
+        <https://www.jstatsoft.org/article/view/v045i03>`_
+    """
+
+    def __init__(
+            self,
+            missing_values='NaN',
+            imputation_order='monotone',
+            n_imputations=100,
+            n_burn_in=10,
+            n_nearest_features=np.infty,
+            initial_fill_method="mean",
+            min_value=-np.infty,
+            max_value=np.infty,
+            verbose=False,
+            random_state=None):
+        from ..linear_model import BayesianRidge  # avoiding circular import
+        self.model = BayesianRidge()  # TODO: need to move to fit_transform?
+        self.missing_values = missing_values
+        self.imputation_order = imputation_order
+        self.n_imputations = n_imputations
+        self.n_burn_in = n_burn_in
+        self.n_nearest_features = n_nearest_features
+        self.initial_fill_method = initial_fill_method
+        self.min_value = min_value
+        self.max_value = max_value
+        self.verbose = verbose
+        self.random_state = random_state
+
+    def _impute_one_feature(self,
+                            X_filled,
+                            mask_missing_values,
+                            this_feature_ind,
+                            other_feature_inds,
+                            model=None,
+                            min_std=1e-6):
+        """
```

Comment:
Follow the notes in docstring from the contributing guide  [here](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)

Reply:
In brief, add short and optional long description, parameters and returns value. Use as much as possible characters up to 79 characters.

Reply:
Will do, thank you.